api/permissions.py

```diff
@@ -27,3 +30,13 @@ def has_permission(self, request, view):
         assert False, ('TokenHasScope requires either the'
                        '`oauth2_provider.rest_framework.OAuth2Authentication` authentication '
                        'class to be used.')
+
+
+class IsDeletedPremissions(BasePermission):
+    """
+    Permission check for deleted objects.
+    """
+    def has_object_permission(self, request, view, obj):
+        if hasattr(obj, 'is_deleted') and obj.is_deleted:
+                raise PermissionDenied('Query is forbidden for the given object.')
```

Comment:
`Query is forbidden` is a bit generic. Go for something with a bit more detail, like `'This {} has been removed.'.format(obj._meta.verbose_name)`. Removed seems a bit softer than deleted.